helper/src/main/java/me/lucko/helper/utils/XMaterial.java

```diff
@@ -0,0 +1,1686 @@
+     * @see #matchXMaterial(String, byte)
+     */
+    @Nullable
+    public static XMaterial getXMaterialIfDuplicated(@Nonnull String name) {
+        Validate.notEmpty(name, "Material name cannot be null or empty");
+        String formatted = format(name);
+        Optional<Map.Entry<XMaterial, XMaterial>> mat = DUPLICATED.entrySet().stream().filter(m
+                -> m.getKey().name().equals(formatted) || m.getValue().name().equals(formatted)).findFirst();
+
+        if (mat.isPresent()) {
+            Map.Entry<XMaterial, XMaterial> found = mat.get();
+            return formatted.equals(found.getKey().name()) ? found.getValue() : found.getKey();
+        }
+        return null;
+    }
+
+    /**
+     * Attempts to build the string like an enum name.
+     *
+     * @param name the material name to modify.
+     * @return a Material enum name.
+     */
+    @Nonnull
+    private static String format(@Nonnull String name) {
+        return StringUtils.replace(StringUtils.replace(name.toUpperCase(), "MINECRAFT:", ""), "-", "_").trim()
+                .replaceAll("\\s+", "_").replaceAll("\\d+", "").replaceAll("\\W+", "");
+    }
+
+    /**
+     * Parses the material name if the legacy name has a version attached to it.
+     *
+     * @param name the material name to parse.
+     * @return the material name with the version removed.
+     */
+    @Nonnull
+    private static String parseLegacyVersionMaterialName(@Nonnull String name) {
+        Validate.notEmpty(name, "Material name cannot be null or empty");
+        if (!name.contains("/")) return name;
+        return name.substring(0, name.indexOf('/'));
+    }
+
+    /**
+     * Checks if the version argument is the same or higher than the current server version.
+     *
+     * @param version the version to be checked.
+     * @return true of the version is equal or higher than the current version.
+     */
+    public static boolean isVersionOrHigher(@Nonnull MinecraftVersion version) {
+        MinecraftVersion current = getVersion();
+
+        if (version == current) return true;
+        if (version == MinecraftVersion.UNKNOWN) return false;
+        if (current == MinecraftVersion.UNKNOWN) return true;
+
+        int ver = Integer.parseInt(StringUtils.replace(StringUtils.replace(version.name(), "VERSION_", ""), "_", ""));
+        int currentVer = Integer.parseInt(StringUtils.replace(StringUtils.replace(current.name(), "VERSION_", ""), "_", ""));
+
+        return currentVer >= ver;
+    }
+
+    /**
+     * Converts the material's name to a string with the first letter uppercase.
+     *
+     * @return a converted string.
+     */
+    @Nonnull
+    public static String toWord(@Nonnull Material material) {
+        return toWord(material.name());
+    }
+
+    /**
+     * Parses a material name like a normal like.
+     * Normal names have underlines removed and each word starts with caps.
```

Comment:
Additional paragraphs in comments should include an empty line between previous paragraphs.